src/dcap_provider.cpp

```diff
@@ -490,26 +546,30 @@ static sgx_plat_error_t build_pck_crl_url(
 
     int crl_size;
     safe_cast(crl_url.size(), &crl_size);
-    std::string escaped = curl_easy::escape(crl_url.data(), crl_size);
+    *out = build_pck_crl_url(crl_url, API_VERSION_LEGACY);
+    return SGX_PLAT_ERROR_OK;
+}
 
-    try
+static std::string build_tcb_info_url(const std::string &fmspc)
+{
+    std::string version = get_collateral_version();
+    std::string client_id = get_client_id();
+    std::stringstream tcb_info_url;
+    tcb_info_url << get_base_url();
+    
+    if (!version.empty())
     {
-        *out = get_base_url() + "/pckcrl?uri=" + escaped + '&';
-
-        client_id = get_client_id();
-        if (!client_id.empty())
-        {
-            *out += "clientid=" + client_id + '&';
-        }
-
-        *out += API_VERSION;
-
-        return SGX_PLAT_ERROR_OK;
+        tcb_info_url << "/" << version;
     }
-    catch (...)

```

Comment:
Has this every thrown an exception in the past? Since it is just building a URL I was thinking it should be fairly bulletproof.

Reply:
I would analyze the function calls within this code block prior to removing any catch statements.

Reply:
I analyzed this already when I pulled it out and nothing was immediately obvious. I thought it was relatively safe to remove since:
- the exception was just being re-thrown (see the removed line 511 below)
- None of the other build_url functions had this

Reply:
Thats fine then, feel free to keep it removed.